dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinUserMetadataRestController.java

```diff
@@ -303,19 +305,87 @@ private void sendEmailWithDownloadLink(Context context, DSpaceObject dso,
             bean.addRecipient(email);
             bean.send();
         } catch (MessagingException e) {
-            log.error("Cannot send the email because: " + e.getMessage());
+            log.error("Cannot send the email with download link, because: " + e.getMessage());
+            throw new MessagingException(e.getMessage());
+        }
+        // If previous mail fails with exception, this block never executes = admin is NOT
+        // notified, if the mail is not really sent (if it fails HERE, not later, e.g. due to mail server issue).
+        sendAdminNotificationEmail(context, downloadLink, dso, clarinLicense, mailType, clarinUserMetadataRestList);
+
+    }
+    enum MailType { ALLZIP, BITSTREAM }
```

Comment:
code style
1. is this the right place?
2. docs

Reply:
moved up, added comment. Is it better?

Reply:
@vidiecan can you resolve this conversation if it is ok now?